playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/ClockSleepTimer.java

```diff
@@ -0,0 +1,143 @@
+package de.danoeh.antennapod.playback.service.internal;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.VibrationEffect;
+import android.os.Vibrator;
+import android.os.VibratorManager;
+import android.util.Log;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import de.danoeh.antennapod.event.playback.PlaybackPositionEvent;
+import de.danoeh.antennapod.event.playback.SleepTimerUpdatedEvent;
+import de.danoeh.antennapod.storage.preferences.SleepTimerPreferences;
+
+public class ClockSleepTimer implements SleepTimer {
+    private static final String TAG = "ClockSleepTimer";
+    private final Context context;
+    private final long initialWaitingTime;
+    private long timeLeft;
+    private boolean isRunning = false;
+    private long lastTick = 0;
+
+    protected boolean hasVibrated = false;
+    protected ShakeListener shakeListener;
+
+    public ClockSleepTimer(final Context context, long initialWaitingTime) {
+        this.context = context;
+        this.initialWaitingTime = initialWaitingTime;
+        this.timeLeft = initialWaitingTime;
+
+        EventBus.getDefault().register(this);
+        EventBus.getDefault().post(SleepTimerUpdatedEvent.justEnabled(initialWaitingTime));
```

Comment:
This sounds a bit risky. I would move this to a more explicit method that is called from the outside. Otherwise we might miss unregistering (when future developers modify the code code) and run into issues. Maybe move to the start method? I also find it a bit unexpected to immediately start something in the constructor